be/src/exec/pipeline/exchange/local_exchange.h

```diff
@@ -188,5 +207,35 @@ class PassthroughExchanger final : public LocalExchanger {
     std::atomic<size_t> _next_accept_source = 0;
 };
 
+// Random shuffle for each chunk of source.
+class RandomPassthroughExchanger final : public LocalExchanger {
+public:
+    RandomPassthroughExchanger(const std::shared_ptr<LocalExchangeMemoryManager>& memory_manager,
+                               LocalExchangeSourceOperatorFactory* source)
+            : LocalExchanger("RandomPassthrough", memory_manager, source) {}
+
+    Status accept(const ChunkPtr& chunk, int32_t sink_driver_sequence) override;
+
+private:
+    RandomPartitioner _random_partitioner;
+};
+
+// When total chunk num is greater than num_partitions, passthrough chunk directyly, otherwise
+// random shuffle each rows for the input chunk to seperate it evenly.
+class AdaptivePassthroughExchanger final : public LocalExchanger {
+public:
+    AdaptivePassthroughExchanger(const std::shared_ptr<LocalExchangeMemoryManager>& memory_manager,
+                                 LocalExchangeSourceOperatorFactory* source)
+            : LocalExchanger("AdaptivePassthrough", memory_manager, source) {}
+
+    Status accept(const ChunkPtr& chunk, int32_t sink_driver_sequence) override;
+
+private:
+    RandomPartitioner _random_partitioner;
+    std::atomic<size_t> _next_accept_source = 0;
+    std::atomic<size_t> _source_total_chunk_num = 0;
+    std::atomic<bool> _is_pass_throgh_by_chunk = false;
```

Comment:
pass_through ?

Reply:
PassThroughExchanger always passes through the input chunk, `AdaptivePassthroughExchanger ` first pass through the input chunk by split chunk meanly, if there are many chunks, just pass though by chunk.

so  use`_is_pass_throgh_by_chunk ` to mark whether pass though by chunk or rows?